triggers/triggers-dt/src/main/java/gg/xp/xivsupport/triggers/Arcadion/M4N.java

```diff
@@ -0,0 +1,126 @@
+	private final ModifiableCallout<AbilityUsedEvent> stampedingThunderWest = new ModifiableCallout<>("Stampeding Thunder: Go East", "Go East");
+	@NpcAbilityUsedCallout(0x92AC)
+	private final ModifiableCallout<AbilityUsedEvent> stampedingThunderEast = new ModifiableCallout<>("Stampeding Thunder: Go West", "Go West");
+
+	public M4N(XivState state) {
+		this.state = state;
+	}
+	private final XivState state;
+
+	@Override
+	public boolean enabled(EventContext context) {
+		return state.dutyIs(KnownDuty.M4N);
+	}
+
+	public static final List<Long> validStacks = List.of(723L,724L);
+
+	public ArenaSector getSafeDirectionForBA(BuffApplied ba) {
+		//723 hits north, go south
+		//724 hits south, go north
+		long stacks = ba.getRawStacks();
+		return stacks == 723 ? ArenaSector.SOUTH : ArenaSector.NORTH;
+	}
+
+	private final ModifiableCallout<?> gunStart = new ModifiableCallout<>("Gun Blasts: Start", "Start {dir}");
+	private final ModifiableCallout<?> gunSafe = new ModifiableCallout<>("Gun Blasts: Next Safe", "{dir}");
+
+	@AutoFeed
+	private final SequentialTrigger<BaseEvent> gunBlast = SqtTemplates.sq(30_000, AbilityCastStart.class,
+			acs -> acs.abilityIdMatches(0x92B0, 0x9B4F, 0x9B56, 0x92AD, 0x9B55, 0x9B57),
+			(e1, s) -> {
+				log.info("Gun Blasts: Start");
+				int iterations = switch((int)e1.getAbility().getId()) {
+					case 0x92B0, 0x92AD -> 2;
+					case 0x9B4F, 0x9B55 -> 3;
+					case 0x9B56, 0x9B57 -> 4;
+					default -> 0;
+				};
+
+				//Buff 0xB9A
+				ArenaSector buff1 = getSafeDirectionForBA(s.waitEvent(BuffApplied.class, ba -> validStacks.contains(ba.getRawStacks()) && ba.buffIdMatches(0xB9A)));
+				log.info("Gun Blasts Start: {}", buff1);
+				s.updateCall(gunStart.getModified(Map.of("dir", buff1)));
+
+				List<BuffApplied> furtherBuffs = s.waitEvents(iterations, BuffApplied.class, ba -> validStacks.contains(ba.getRawStacks()) && ba.buffIdMatches(0xB9A));
+
+				boolean wasNorth = buff1 != ArenaSector.SOUTH;
+				//repeats up to 4 times
+				for(int i = 0; i < iterations; i++) {
+					//many IDs. Some may be missing. Ability name "Wicked Cannon"
+					//0x4E40, 0x9BAC, 0x92AE, 0x9BBE, 0x92AF
+					s.waitEvent(AbilityUsedEvent.class, aue -> aue.abilityIdMatches(0x4E40, 0x9BAC, 0x92AE, 0x9BBE, 0x92AF) && aue.isFirstTarget());
+					ArenaSector safeSide = getSafeDirectionForBA(furtherBuffs.get(i));
+					log.info("Gun Blasts {}: {}", i, safeSide);
+					if(wasNorth && safeSide == ArenaSector.SOUTH) {
+						s.updateCall(gunSafe.getModified(Map.of("dir", ArenaSector.SOUTH)));
+						wasNorth = false;
+					}
+					else if(!wasNorth && safeSide == ArenaSector.NORTH) {
+						s.updateCall(gunSafe.getModified(Map.of("dir", ArenaSector.NORTH)));
+						wasNorth = true;
+					}
+				}
+			});
+
+	public static final List<Long> validIDs = List.of(793L, 794L);
+
+	private final ArenaPos arenaPos = new ArenaPos(100, 100, 5, 5);
+
+	private final ModifiableCallout<?> shadowSafe = new ModifiableCallout<>("Shadow's Sabbath", "{dir}");
```

Comment:
There isn't a good indication here of whether {dir} is referring to the safe direction, or the unsafe direction. I like to do something like `{safe} safe` so that both the callout and variable name are clear.